src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllProvider.cs

```diff
@@ -39,5 +40,31 @@ public virtual IEnumerable<string> GetSupportedFixAllDiagnosticIds(CodeFixProvid
         /// Gets fix all occurrences fix for the given fixAllContext.
         /// </summary>
         public abstract Task<CodeAction> GetFixAsync(FixAllContext fixAllContext);
+
+        internal virtual Task<ImmutableDictionary<Document, ImmutableArray<Diagnostic>>> GetDocumentDiagnosticsToFixAsync(FixAllContext fixAllContext)
```

Comment:
We can get rid of GetDocumentDiagnosticsToFixAsync and GetProjectDiagnosticsToFixAsync being added here. The only place them seem to be invoked are in the Suppression batch fixer, which explicitly uses BatchFixAllProvider.